website_sale_affiliate/models/sale_affiliate_request.py

```diff
@@ -0,0 +1,86 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 LasLabs Inc.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl)
+
+from datetime import timedelta
+
+from odoo import api, fields, models
+from odoo.http import request
+
+
+class AffiliateRequest(models.Model):
+    _name = 'sale.affiliate.request'
+    _order = 'create_date desc'
+
+    name = fields.Char(required=True, index=True)
+    affiliate_id = fields.Many2one(
+        'sale.affiliate',
+        string='Affiliate',
+        required=True,
+        help='Affiliate that referred request',
+    )
+    date = fields.Datetime(
+        string='Start Date',
+        required=True,
+        default=lambda self: fields.Datetime.now(),
+        help='Date and time of initial request',
+    )
+    ip = fields.Char(
+        string='Client IP',
+        required=True,
+        default=lambda self: request.httprequest.headers.environ.get(
+            'REMOTE_ADDR',
+        ),
+    )
+    referrer = fields.Char(
+        default=lambda self: request.httprequest.headers.environ.get(
+            'HTTP_REFERER',
+        ),
+        help='Request session referrer header',
+    )
+    user_agent = fields.Char(
+        required=True,
+        default=lambda self: request.httprequest.headers.environ.get(
+            'HTTP_USER_AGENT',
+        ),
+        help='Request session user agent',
+    )
+    accept_language = fields.Char(
+        required=True,
+        default=lambda self: request.httprequest.headers.environ.get(
+            'HTTP_ACCEPT_LANGUAGE',
+        ),
+        help='Request session accept language',
+    )
+    sale_ids = fields.One2many(
+        'sale.order',
+        'affiliate_request_id',
+        string='Sales',
+        help='Qualified conversions generated as a result of affiliate request'
+    )
+
```

Comment:
We need to save the affiliate reference too

Reply:
The `aff_key` kwarg is stored in the affiliate request's `name` field. I added a help parameter to the field for clarity.